### ISSUE TITLE
pairwise_dist eps argument is confusing

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#30263]
The doc says that `eps` helps preventing division by zero (without mentioning how).  The actual implementation is 
https://github.com/pytorch/pytorch/blob/0fdbb762d1a9e6b9c262760b6b67bcb3d0814407/aten/src/ATen/native/Distance.cpp#L15-L17

However, `norm(zero_vector)` is perfectly stable and gives correct gradient. So unless `pairwise_dist` output is used as a denominator, there is no DBZ or any instability. And how to use `pairwise_dist` output should be in users' control, unless we want to automatically add `eps` to every op that can output zeros.

**Comment by cpuhrsch** [comment]
I'm also in favor of this. This flag effectively makes the user responsible for numerical stability and it's not consistent. I'll flag it for triage review so we can make a decision.

**Comment by vadimkantorov** [comment]
Is `norm` special-cased to be stable at zero?

**Comment by albanD** [comment]
Yes, we use the subgradient of 0 at 0 for norm since https://github.com/pytorch/pytorch/pull/2775

**Comment by ezyang** [comment]
Seems reasonable to me. We could start by modifying the documentation.
